google/cloud/aiplatform/models.py

```diff
@@ -852,7 +854,42 @@ def predict(self, instances: List, parameters: Optional[Dict] = None) -> Predict
     # TODO(b/172828587): implement prediction
     def explain(self, instances: List[Dict], parameters: Optional[Dict]) -> List[Dict]:
         """Online prediction with explanation."""
-        raise NotImplementedError("Prediction not implemented.")
+        raise NotImplementedError("Prediction with explanation not implemented.")
+
+    def list_models(self) -> Sequence[gca_endpoint.DeployedModel]:
+        """Returns a list of the models deployed to this Endpoint.
+
+        Returns:
+            deployed_models (Sequence[aiplatform.gapic.DeployedModel]):
+                A list of the models deployed in this Endpoint.
+        """
+        self._sync_gca_resource()
+        return self._gca_resource.deployed_models
+
+    def undeploy_all(self):
```

Comment:
Calling the `_undeploy` method skips unnecessary argument validation for this case.

Additionally, it looks like this can be made optionally sync, by adding a sync parameter and passing it to `_undeploy`. No need to decorate the method. The undeploys will queue on the endpoint object.